BlueprintUI/Sources/Element/ElementContent.swift

```diff
@@ -16,40 +16,67 @@ public struct ElementContent {
         measure(
             in: constraint,
             environment: environment,
-            cache: CacheFactory.makeCache(name: "ElementContent"),
+            cacheName: "ElementContent",
             layoutMode: RenderContext.current?.layoutMode ?? environment.layoutMode
         )
     }
 
     func measure(
         in constraint: SizeConstraint,
         environment: Environment,
-        cache: CacheTree,
+        cacheName: String,
         layoutMode: LayoutMode
     ) -> CGSize {
-        // TODO: switch on layoutMode
-        storage.measure(in: constraint, environment: environment, cache: cache)
+        switch layoutMode {
+        case .legacy:
+            return measure(
+                in: constraint,
+                environment: environment,
+                cache: CacheFactory.makeCache(name: cacheName)
+            )
+        case .caffeinated:
+            return sizeThatFits(
+                proposal: constraint,
+                context: MeasureContext(environment: environment)
+            )
+        }
     }
 
     func measure(in constraint: SizeConstraint, environment: Environment, cache: CacheTree) -> CGSize {
         storage.measure(in: constraint, environment: environment, cache: cache)
     }
 
+    func sizeThatFits(proposal: SizeConstraint, context: MeasureContext) -> CGSize {
+        storage.sizeThatFits(proposal: proposal, context: context)
+    }
+
     public var childCount: Int {
         storage.childCount
     }
 
-    func performLayout(
+    typealias IdentifiedNode = (identifier: ElementIdentifier, node: LayoutResultNode)
```

Comment:
This is probably something to clean up later; but I found it much nicer (and much better for debugging) in the caching branch by removing this tuple and putting the ID onto `LayoutResultNode` itself directly.